### DISCUSSION TITLE
Concatenate multiple environment variable

### DISCUSSION THREAD

**Discussion by jy95** [community/community#26208] [Actions]
<p>Hello,</p>
<p>The ORM I use ( Sequelize ) needs to get my database connection URL into a env variable  <strong>DATABASE_URL </strong></p>
<p>For example : </p>
<pre><code>constsequelize=newSequelize('postgres://user:pass@example.com:5432/dbname')// Example for postgres
</code></pre>
<p>Now, I would like to have this kind of  with my action  : How can I do that ?</p>
<pre><code>name: Source Code CI/CD

on:
push:
branches:
- master
pull\_request:
branches:
- master

jobs:
ci:
runs-on: ubuntu-latest
container:
image: node
services:
# More explanation about that here :  
# https://github.com/actions/example-services/blob/master/.github/workflows/postgres-service.yml  
postgres:
image: postgres:12-alpine
env:
POSTGRES\_USER: postgres
POSTGRES\_PASSWORD: jy95
POSTGRES\_DB: sourcecode
ports: ["5432:5432"]
options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5

steps:
-uses: actions/checkout@v1
-name: Install &amp; Tests
env:
# use postgres for the host here because we have specified a container for the job.  
 # If we were running the job on the VM this would be localhost  
POSTGRES\_HOST: postgres
POSTGRES\_PORT: ${{ job.services.postgres.ports[5432] }}
# TODO How to concatenate multiple env variables in Github Actions ?  
DATABASE\_URL: ???
run: |
npm install
npm test
cd:
runs-on: ubuntu-latest
needs: ci

steps:
-uses: actions/checkout@v1
-name: Docker login
run: docker login -u ${{ secrets.DOCKER_USER }} -p ${{ secrets.DOCKER_PASSWORD }}
-name: Build
run: docker build -t sourcecode_api .
-name: Tags
run: |
docker tag back ${{ secrets.DOCKER_USER }}/sourcecode_api:${{ github.sha }}
docker tag back ${{ secrets.DOCKER_USER }}/sourcecode_api:latest
-name: Push
run: |
docker push ${{ secrets.DOCKER_USER }}/sourcecode_api:${{ github.sha }}
docker push ${{ secrets.DOCKER_USER }}/sourcecode_api:latest
</code></pre>
<p>Thanks in advanced</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>You could set those env which need to use in <strong>DATABASE_URL </strong> as job environment variables. </p>
<p>For POSTGRES_PORT, we could use another step to set it as an env. </p>
<p>Then to get those env value, you can use ${{ env.POSTGRES_***}} syntax to concatenate multiple env variables. </p>
<pre><code>DATABASE_URL: '${{ env.POSTGRES_HOST }}://${{ env.POSTGRES_USER}}:${{env.POSTGRES_PASSWORD}}@example.com:${{env.POSTGRES_PORT}}/${{env.POSTGRES_DB}}'
</code></pre>
<p>Please see my example yml content: </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/2b6e23c63f797b8ef593412a43198e75a6cebd7c.png" data-download-href="/uploads/short-url/6ccvhsuOnuiE2c1MTY6xHwCIiF6.png?dl=1" title="env in other env.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181096060-6f7fa369-1ab9-4370-a06d-f5f76ce745b1.png" alt="env in other env.png" data-base62-sha1="6ccvhsuOnuiE2c1MTY6xHwCIiF6" width="515" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/2/2b6e23c63f797b8ef593412a43198e75a6cebd7c_2_515x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/2/2b6e23c63f797b8ef593412a43198e75a6cebd7c_2_772x750.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/2/2b6e23c63f797b8ef593412a43198e75a6cebd7c.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/2b6e23c63f797b8ef593412a43198e75a6cebd7c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">env in other env.png</span><span class="informations">946×918 27.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p> Here is the result: </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/9/9c8fab91afe899282965f43d73735c1d2ab68506.png" data-download-href="/uploads/short-url/ml0ir2klOtRcwfBUZKx9p5jBGC2.png?dl=1" title="result.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181096061-f04f8ca0-ec54-4aec-9772-fef8339c0b1f.png" alt="result.png" data-base62-sha1="ml0ir2klOtRcwfBUZKx9p5jBGC2" width="690" height="483" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/9/9c8fab91afe899282965f43d73735c1d2ab68506_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">result.png</span><span class="informations">807×565 14.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by jy95** [comment]
<p><a class="mention" href="/yanjingzhu">@yanjingzhu</a> Thanks for the help.</p>
<p>The concatenation works but I still cann’t access the database ( <a href="https://stackoverflow.com/questions/60321175/github-actions-unhandled-rejection-sequelizeconnectionerror-no-postgresql-use" rel="nofollow noopener">https://stackoverflow.com/questions/60321175/github-actions-unhandled-rejection-sequelizeconnectionerror-no-postgresql-use</a> )</p>
